drivers/serial/uart_mspm0g3xxx.c

```diff
@@ -81,6 +81,10 @@ static void uart_mspm0g3xxx_poll_out(const struct device *dev, unsigned char c)
 }
 
 #ifdef CONFIG_UART_INTERRUPT_DRIVEN
+
+#define UART_MSPM0_TX_INTERRUPTS (DL_UART_INTERRUPT_TX | DL_UART_INTERRUPT_EOT_DONE)
+#define UART_MSPM0_RX_INTERRUPTS (DL_UART_INTERRUPT_RX)
```

Comment:
I'd have added these on the commit where you introduced checking `EOT_DONE` as well